### DISCUSSION TITLE
Noise Removal

### DISCUSSION THREAD

**Discussion by Hanysabeh** [ouster-lidar/ouster_example#582] [Q&A] [answered]
Is ouster library removing the noise automatically when it read a pcap or a live stream from sensor or i will need to remove the noise by another libraries and what if i want to remove the static points is that supported by ouster library ?

Thanks in advance

**Comment by Samahu** [comment, marked as answer]
The sensor FW does most of the work to eliminate and reduce noise, however, there can be certain scenarios with certain weather condition or signal interference that can generate some residual noise. In this case you would need to build your own routines to cancel this type of  noise. We do have two classes `AutoExposure` and `BeamUniformityCorrector` that can be helpful when trying to visualize certain channels.
